python/ray/data/context.py

```diff
@@ -283,6 +284,15 @@ def _set_current(context: "DataContext") -> None:
         global _default_context
         _default_context = context
 
+    def get_plugin_config(self, key: str, default: Any = None) -> Any:
+        return self._plugin_configs.get(key, default)
+
+    def set_plugin_config(self, key: str, value: Any) -> None:
+        self._plugin_configs[key] = value
+
+    def remove_plugin_config(self, key: str) -> None:
+        self._plugin_configs.pop(key, None)
```

Comment:
let's declare these methods with `_` prefix, so they remain private and easy to change later.

Reply:
Discussed offline, we'll make this API reusable for other components.